client/base/Router.js

```diff
@@ -1,21 +1,13 @@
-import React, { PropTypes } from 'react'
+import React from 'react'
 import { Router as ReactRouter } from 'react-router'
-import { FixNamedRoutesSupport as allowNames } from 'react-router-named-routes'
+import { syncHistoryWithStore } from 'react-router-redux'
 
-import { appRoutes } from 'modules/app'
+import { appRoutes as routes } from 'modules/app'
 
-allowNames(appRoutes)
-
-const propTypes = {
-  history: PropTypes.object.isRequired
-}
-
-export default function Router({ history }) {
+export default function Router({ store, history }) {
   return (
-    <ReactRouter history={history}>
-      {appRoutes}
-    </ReactRouter>
+    <ReactRouter
+        history={syncHistoryWithStore(history, store)}
```

Comment:
Can this be moved back where it was in `client/index.js`?  Then we wouldn't need to pass the `store` around.